src/bin/pg_autoctl/monitor.c

```diff
@@ -346,6 +358,217 @@ monitor_node_active(Monitor *monitor,
 }
 
 
+/*
+ * monitor_set_node_replication_settings updates the monitor on the changes
+ * in the node replication settings. Caller should first retrieve
+ * existing values if setting only one parameter.
```

Comment:
Why build the API that way? It's fragile. We could maybe use the classic Common Lisp trick for function keyword parameters where for each parameter there's an extra bit of information: parameter-name-s-p, where s-p stands for supplied-predicate. It is set to true only when a specific value has been passed through for this parameter. That allows to pass NULL and say “yeah I intentionally passed NULL, it's not the default”.

So we could have two extra bool parameters here, set_candidate_priority and set_replication_quorum, that would only be true when that's what we want to do.

Another way to do it would be to write two functions, one per parameter. Do we have any call-sites that are setting both the values in a single call?